include/envoy/http/codec.h

```diff
@@ -189,6 +210,9 @@ struct Http2Settings {
   // our default connection-level window also equals to our stream-level
   static const uint32_t DEFAULT_INITIAL_CONNECTION_WINDOW_SIZE = 256 * 1024 * 1024;
   static const uint32_t MAX_INITIAL_CONNECTION_WINDOW_SIZE = (1U << 31) - 1;
+
+  // By default, do not enforce stream buffer limits beyond the connection level limits.
+  static const uint32_t DEFAULT_PER_STREAM_BUFFER_LIMIT = 0;
```

Comment:
Current defaults IIRC are 1MiB for TCP level buffering, but 256MiB for both connection and stream HTTP/2 windows. We can tweak defaults in follow up (or here), but my main question is actually why to have the stream window default be different from stream buffer limit. Seems like they could be the same?